modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/jwt/JoseJwt.kt

```diff
@@ -0,0 +1,4 @@
+package com.sphereon.oid.fed.common.jwt
+
+expect fun sign(payload: String, header: String, opts: Map<String, Any>): String
```

Comment:
I do not like this whole  signature.
First of all, opts: Map<String, Any>, with this you are putting the platform differences with the implementer. Please create a static options data class with options that work for all platforms. Do the mapping in the platform implementation. If there are options available on one platform that are not available on the other we should think hard if we really need to expose them or if we can omit them or set a default value that works for us.

Then payload: String, header: String is also not very clear. Make it at least 
jwtHeader:String, jwtPayload:String or better, create a little data class
data class UnsignedJWT(val header:String, val payload,String) and use that as parameter.

Reply:
See the discussion in slack about the opts. They are they are there for external implementors. Not for platform specifics. 

Regarding the header and payload. How did that ever change to string. That doesn't make any sense. These should be objects, that are translated into strings in this exact function

Reply:
Changes implemented